record-builder-core/src/main/java/io/soabase/recordbuilder/core/RecordBuilder.java

```diff
@@ -378,6 +388,18 @@ enum ConcreteSettersForOptionalMode {
         DISABLED, ENABLED, ENABLED_WITH_NULLABLE_ANNOTATION,
     }
 
+    enum GeneratedAnnotation {
```

Comment:
If we use this enum I'd like to think of better names. `ALWAYS` and `NEVER` are limiting. We will have trouble add another values that works with these. Some alternatives:

- `ENABLED` / `DISABLED`
- `INCLUDE` / `EXCLUDE`

Also, we could name the method `generatedAnnotationMode` or something similar.